### PR TITLE
sgfutils: init at 0.25-unstable-2017-11-27

### PR DESCRIPTION
**Pull request by ggPeti**
## Description of changes

Adds [sgfutils](https://homepages.cwi.nl/~aeb/go/sgfutils/html/sgfutils.html) package - [someone is writing a book with the help of this](https://www.reddit.com/r/baduk/comments/185087q/comment/kb6jqjm/?utm_source=reddit&utm_medium=web2x&context=3).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

note from the PR author: I checked the release notes checkboxes because each implication is vacuously true i.e. this change is just a leaf package, not a major or breaking change, and not a module

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/sgfutils/default.nix
Comment (2xsaiko) [inline pkgs/games/sgfutils/default.nix:32]: Wouldn't it be better to patch the source file to specify the full path of `convert` instead of wrapping?

Comment (ggPeti) [reply pkgs/games/sgfutils/default.nix:32]: it wouldn't IMO. patches are more fragile - if the source ever changes, which it admittedly might not, then it becomes a chore to change the patch as well.

File: pkgs/games/sgfutils/default.nix
Comment (mweinelt) [inline pkgs/games/sgfutils/default.nix:21]:
```suggestion
  buildPhase = ''
    runHook preBuild
    make all
    runHook postBuild
  '';
```

When overriding a standard phase, make sure to call the pre and pots hooks. Same review for `installPhase`.

File: pkgs/top-level/all-packages.nix
Comment (mweinelt) [inline pkgs/top-level/all-packages.nix:38338]: Move this package into `pkgs/by-name/sg/sgfutils/package.nix` instead and drop the change to this file.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md

File: pkgs/games/sgfutils/default.nix
Comment (mweinelt) [inline pkgs/games/sgfutils/default.nix:14]: Odd choice to put `buildInputs` up here, and not closer to `buildPhase`.

File: pkgs/games/sgfutils/default.nix
Comment (mweinelt) [inline pkgs/games/sgfutils/default.nix:13]: > If a package is a commit from a repository without a version assigned, then the version attribute should be the latest upstream version preceding that commit, followed by -unstable- and the date of the (fetched) commit. The date must be in "YYYY-MM-DD" format.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

Comment (ggPeti) [reply pkgs/games/sgfutils/default.nix:13]: there is no upstream version before the commit, and there are 1526 instances of `unstable-YYYY-MM-DD` versions throughout the repo. what do you suggest?

File: pkgs/games/sgfutils/default.nix
Comment (2xsaiko) [reply pkgs/games/sgfutils/default.nix]: I'm talking about substituteInPlace, you don't have to change that unless they change the binary name. Obviously a diff patch is unreasonable, especially because you'd have to generate it from a template first.

File: pkgs/games/sgfutils/default.nix
Comment (mweinelt) [reply pkgs/games/sgfutils/default.nix]: Remains unresolved.

Comment (ggPeti) [reply pkgs/games/sgfutils/default.nix]: I understand your opinion and I disagree with it. As there is no contributing stipulation that governs where buildInputs is, I consider it resolved.

File: pkgs/games/sgfutils/default.nix
Comment (ggPeti) [reply pkgs/games/sgfutils/default.nix]: The same point stands with substituteInPlace. Why define a brittle patch when a robust wrapper gets the same thing done?

File: pkgs/games/sgfutils/default.nix
Comment (ggPeti) [reply pkgs/games/sgfutils/default.nix]: Ok, added 0.25 as the github revision only contains some extra files compared to the tarball, no differences in existing ones.

**Comment by ggPeti** [comment]
@ajs124 why did you mark the PR as draft this time?

**Comment by ajs124** [comment]
Because for a split second a thought I cared about not having bad code and commits in nixpkgs, but then I remembered that I don't. Don't ping me again.

**Comment by ggPeti** [comment]
requesting a contributor to remove the above inappropriate, toxic comment along with this one. we don't need any of this.

File: pkgs/by-name/sg/sgfutils/package.nix
Comment (flokli) [inline pkgs/by-name/sg/sgfutils/package.nix:42]: Is this now a music similarity library or tools to deal with go games?

File: pkgs/games/sgfutils/default.nix
Comment (2xsaiko) [reply pkgs/games/sgfutils/default.nix]: I suppose so, I just think that's generally preferable over the env leakage and messed up argv[0] caused by wrappers, which can both cause issues in other packages down the line. The Qt wrapper has bitten me just a few days ago by Konsole's (or something else?) setting QT_PLUGIN_PATH which caused me to miss some dependencies of something I was packaging because they were already in the inherited environment, and I only caught it because I tested the package on my Mac afterwards.

Though tbf in reality both are unlikely to cause a big problem here other than an ugly name in the help output (if even) and the process list.

Comment (ggPeti) [reply pkgs/games/sgfutils/default.nix]: anyway I just realized that even if `convert` was substituted in place, the FONTCONFIG_FILE env var should be set.

**Review by dblsaiko — APPROVED** [review]

File: pkgs/games/sgfutils/default.nix
Comment (2xsaiko) [reply pkgs/games/sgfutils/default.nix]: Ah yeah, that's true.

**Comment by ggPeti** [comment]
Squashed and rebased on latest master.

**Comment by ggPeti** [comment]
>The branch this PR will merge in to does not cleanly evaluate, and so this PR cannot be checked.

retriggering eval with a new force push

File: pkgs/games/sgfutils/default.nix
Comment (turboMaCk) [reply pkgs/games/sgfutils/default.nix]: I agree this is minor and frankly I would probably not request this myself to be honest. Anyway as a gesture of good will I would suggest we follow @mweinelt's suggestion as a gesture of a good will since he did most of the work for reviewing this PR.

@ggPeti Could you move `buildInputs` above the `buildPhase`. I think we ideally want to make everyone involved happy with the merge.

**Comment by turboMaCk** [comment]
Thanks @ggPeti. Anyway to keep everyone happy I would suggest we also address this comment https://github.com/NixOS/nixpkgs/pull/271166#discussion_r1414506919

So my suggestion is:

1. Please move `buildInputs` following the suggestion
2. Squash commits in the PR to one again
3. I think we can merge this

I have a feeling that communication in this PR was not as smooth as I think all parties involved would hope for. This is my suggestion for best resolution.

Once this is done I will be happy to merge PR myself with peace in mind that no one has a reason to feel wronged in the process. I hope this is acceptable to everyone.

File: pkgs/by-name/sg/sgfutils/package.nix
Comment (turboMaCk) [inline pkgs/by-name/sg/sgfutils/package.nix:20]: sorry one more thing. I think `makeWrapper` should be part of `nativeBuildInputs` rather than build inputs.

File: pkgs/games/sgfutils/default.nix
Comment (ggPeti) [reply pkgs/games/sgfutils/default.nix]: Certainly. I believe there's something to say for the role of maintainers though. It's a stylistic difference that is very likely to only matter to the person maintaining the package (me). To unresolve my decision and assert dominance over this insubstantive matter is quite petty and unproductive. But I'm down to make people happy.

**Comment by turboMaCk** [comment]
- [x] Builds locally on linux x86_64
- [x] Binaries run
- [x] All review requests seems to be resolved

**Pull request merged**
